provider/kubernetes.go

```diff
@@ -145,29 +153,22 @@ func (provider *Kubernetes) loadIngresses(k8sClient k8s.Client) (*types.Configur
 						}
 					}
 				}
-				if len(pa.Path) > 0 {
-					ruleType := i.Annotations["traefik.frontend.rule.type"]
 
-					switch strings.ToLower(ruleType) {
-					case "pathprefixstrip":
-						ruleType = "PathPrefixStrip"
-					case "pathstrip":
-						ruleType = "PathStrip"
-					case "path":
-						ruleType = "Path"
-					case "pathprefix":
-						ruleType = "PathPrefix"
-					case "":
-						ruleType = "PathPrefix"
-					default:
-						log.Warnf("Unknown RuleType %s for %s/%s, falling back to PathPrefix", ruleType, i.ObjectMeta.Namespace, i.ObjectMeta.Name)
-						ruleType = "PathPrefix"
+				if len(pa.Path) > 0 {
+					ruleType, unknown := getRuleTypeFromAnnotation(i.Annotations)
+					switch {
```

Comment:
Not a dumb question **at all!**

I agree with you that there should usually be a `default` case when you use a switch statement in an _exhaustive way_: Handle all acceptable inputs and do something for the unexpected case (like returning an error).

The [other type of switch usage](https://golang.org/doc/effective_go.html#switch) in Go is a more readable version of multiple / nested / complex if-else statements. This one doesn't necessarily need a `default` case in my opinion, and it's the one I had in mind here.

Admittedly, I could have also written

```go
if unknown {
  log.Warnf("Unknown RuleType ...")
}

if ruleType == "" {
  ruleType = ruleTypePathPrefix
}
```

I prefer the switch variant because I can use `fallthrough` to shortcut right to setting a default value in case of an unknown rule type (`fallthrough` ignores the condition of the following case expression) and don't need to depend on`ruleType` having a particular value (empty string here).

If you think that two if statements would be better, I'm happy to change.